spec/amp-cors-requests.md

```diff
@@ -37,7 +37,9 @@ by default.
 
 ## CORS Security in AMP
 
-This security protocol consists of two components: CORS origin and source origin restrictions.
+This security protocol consists of three components: `AMP-Same-Origin`, CORS origin and source origin restrictions.
+
+On same origin requests, AMP will set `AMP-Same-Origin: true` custom header. If this header is set it indicates the request is coming from same origin.
```

Comment:
Why check Origin first? The end result would obviously be the same, but checking AMP-Same-Origin first sounds more logical to me.

For same origin requests the AMP-Same-Origin check will always succeed while for real legitimate CORS, Origin+source origin check will always succeed. When using the other ordering, for same Origin requests, the code path depends on browser, method, etc.